src/tools/lifter/bir_lifting_machinesLib_instances.sml

```diff
@@ -589,5 +609,327 @@ val _ = assert bmr_rec_sanity_check (m0_mod_bmr_rec_BigEnd_Main)
 val _ = assert bmr_rec_sanity_check (m0_mod_bmr_rec_LittleEnd_Main)
 
 
+(****************************)
+(* Instantiation for RISC-V *)
+(****************************)
+(* TODO: bmr_normalise_step_thm and bytes_of_hex_code are defined
+ * above - make separate variants for RISC-V? *)
+
+
+(* This is the riscv_step_hex' function. It is based on
+ * riscv_step_hex from l3-machine-code/riscv, and it is stored
+ * in the BIR machine record, to be used to obtain effects of
+ * computational steps. *)
+
+(* WRONG: We will use the regular riscv_step_hex
+ * until bir_nzcv_introsScript, riscv_stepScript,
+ * bir_riscv_extrasScript and bir_lifting_machinesScript can be
+ * modified to accommodate the RISC-V model. *)
+
+(* Type rewrites as a list of theorems (ARM8 also had rewrites
+ * for ``:ProcState``)... *)
+val riscv_REWRS = (
+  (type_rws ``:riscv_state``)
+);
+(* From M0:
+val m0_REWRS = (RName_distinct :: (
+   (type_rws ``:m0_state``) @
+   (type_rws ``:PSR``) @ (* From m0Script *)
+   (type_rws ``:RName``) @ (* From m0Script *)
+   (type_rws ``:Mode``) (* From m0Script *)
+));
+*)
+(* ... and as a simplification set. *)
+val riscv_extra_ss = rewrites (riscv_REWRS@[combinTheory.APPLY_UPDATE_THM])
+
+local
+  (* The naming convention for this is slightly different in the
+   * RISC-V version of the HOL4 model. *)
+  val next_state_tm =
+    prim_mk_const{Name="NextRISCV", Thy="riscv_step"}
+
+  (* TODO: nzcv_FOLDS_RISCV would be defined in
+   * bir_nzcv_introsScript.sml in this directory, if it existed.
+   * ExtendValue_0 does not exist in riscv_stepTheory. No idea
+   * why it is useful. *)
+(* From ARM8:
+  val simp_conv = (SIMP_CONV std_ss [nzcv_FOLDS_ARM8] THENC
+                   SIMP_CONV std_ss [arm8_stepTheory.ExtendValue_0,
+                                     arm8_extra_FOLDS]);
+*)
+  val simp_conv = SIMP_CONV std_ss [riscv_extra_FOLDS]
+
+  (* TODO: word_add_to_sub_TYPES is from bir_arm8_extrasTheory. *)
+(* From ARM8:
+  val simp_conv2 = (SIMP_CONV (arith_ss++wordsLib.WORD_ARITH_ss++
+                               wordsLib.WORD_LOGIC_ss) []
+                   ) THENC
+                   (SIMP_CONV std_ss
+                              [word_add_to_sub_TYPES,
+                               alignmentTheory.aligned_numeric,
+                               wordsTheory.WORD_SUB_INTRO,
+                               wordsTheory.WORD_MULT_CLAUSES]
+                   );
+*)
+  val simp_conv2 =
+    (SIMP_CONV (arith_ss++wordsLib.WORD_ARITH_ss++
+                wordsLib.WORD_LOGIC_ss) []
+    ) THENC
+    (SIMP_CONV std_ss
+               [bir_riscv_extrasTheory.word_add_to_sub_TYPES,
+                alignmentTheory.aligned_numeric,
+                wordsTheory.WORD_SUB_INTRO,
+                wordsTheory.WORD_MULT_CLAUSES]
+    )
+
+  (* TODO: bmr_extra_ARM8 is from bir_lifting_machinesTheory. *)
+(* From ARM8:
+  fun riscv_extra_THMS vn = let
```

Comment:
Here it is probably more clear what I meant in the last comment. If I don't see it wrong, then this comment starts with "From ARM8" and it is even the exact same like the function definition that follows after the comment. Why did you keep this?

Reply:
Yeah, more old TODO items and comments which had not been cleaned out. Fixed now.